### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by lubej**
- Add MetaMask install button, in case MetaMask is not installed;
- Improve edge cases;
- remove infinite loops;

### CODE REVIEW

File: frontend/src/stores/ethereum.ts [hunk 1]
```diff
@@ -122,8 +152,14 @@ export const useEthereumStore = defineStore('ethereum', () => {
...
-                chainName: 'Sapphire Mainnet',
-                rpcUrls: ['https://sapphire.oasis.io'],
+                chainName: 'Oasis Sapphire',
+                nativeCurrency: {
+                  name: 'ROSE',
+                  symbol: 'ROSE',
+                  decimals: 18,
+                },
+                rpcUrls: ['https://sapphire.oasis.io/', 'wss://sapphire.oasis.io/ws'],
+                blockExplorerUrls: ['https://explorer.sapphire.oasis.io'],
```
Comment (buberdds) [inline frontend/src/stores/ethereum.ts]: Can we switch urls to an official explorer in both `blockExplorerUrls`?

File: frontend/src/components/AccountPicker.vue [hunk 2]
```diff
@@ -41,9 +50,14 @@ const isXlScreen = useMedia('(min-width: 1280px)');
         }}</span>
       </span>
     </span>
+    <span class="account-picker-content" v-else-if="!isMetaMaskInstalled">
```
Comment (buberdds) [inline frontend/src/components/AccountPicker.vue:71]: parent should be a button

**Review by buberdds — APPROVED** [review]

File: frontend/src/components/AccountPicker.vue [hunk 3]
```diff
@@ -5,23 +5,32 @@ import { Network, networkName, useEthereumStore } from '../stores/ethereum';
...
 
 const netName = computed(() => networkName(eth.network));
 const unkNet = computed(() => eth.network === Network.Unknown);
 
 const connecting = ref(false);
-const showingConnecting = ref(false);
+const isMetaMaskInstalled = ref(true);
 
 async function connectWallet() {
+  if (!isMetaMaskInstalled.value) {
```
Comment (matevz) [inline frontend/src/components/AccountPicker.vue:19]: Can you also fix the mouse pointer resembling something clickable over the connect wallet button.

image

File: frontend/src/views/HomeView.vue [hunk 4]
```diff
@@ -56,6 +55,8 @@ async function fetchProposals(
 }
 
 onMounted(async () => {
+  await eth.switchNetwork(Network.FromConfig);
```
Comment (matevz) [inline frontend/src/views/HomeView.vue:60]: Not sure why, but this didn't trigger the network switch from Sapphire Mainnet (using it previously for sth else) to Hardhat Localnet (having it set in .env.development). It triggered the switch once I clicked on the new poll button (in PollView.vue). Can you take a look?

File: frontend/src/components/AccountPicker.vue [hunk 5]
```diff
@@ -41,9 +50,14 @@ const isXlScreen = useMedia('(min-width: 1280px)');
         }}</span>
       </span>
     </span>
+    <span class="account-picker-content" v-else-if="!isMetaMaskInstalled">
+      <span>
+        <span>Install MetaMask</span>
```
Comment (matevz) [inline frontend/src/components/AccountPicker.vue:73]: Ideally this should be shown immediately in place of the "Connect Wallet" button, if Metamask is not installed. Using Chrome locally without Metamask, first, the "Connect Wallet" was shown and after clicking on it, the "Install Metamask" button was shown instead.

**Review by matevz — CHANGES_REQUESTED** [review]
I tested this locally and there a still some edges to be polished.

**Comment by matevz** [comment]
FYI gasless voting was merged into master branch. Please rebase.

File: frontend/src/views/HomeView.vue (hunk 4 above)
Comment (lubej) [reply frontend/src/views/HomeView.vue:60]: Sometimes the notification from MetaMask gets hidden, for the network change request. Did add the hard reload for the page to ensure that data gets refresh after network change, let me know if it works for you.

**Review by matevz — APPROVED** [review]
Good job. LGTM. FYI I already deployed this to https://playground.oasis.io/demo-voting

**Pull request merged**